### ISSUE TITLE
2021 Status Update

### ISSUE THREAD

**Issue by NoelDeMartin** [NoelDeMartin/solid-focus#10]
**TLDR: I am planning to update this app at some point, but it hasn't been updated for a while so it may not work with some new PODs. I still use it every day, so if it works for you I encourage you to keep using it.**

I published this app 2.5 years ago, and it was my first Solid App. Since then, I have made other apps and I have learned a lot more about Solid. So I am planning to update this with everything I've learned.

These are some of the things I'm looking forward to do:

- Update the authentication mechanism to work with newer PODs.
- Redefine the data schema (I will provide a user friendly migration path for existing data).
- Use an offline-first approach.
- Improve caching.
- Improve error handling.

Having said that, I don't want to discourage anyone from using the app if it works for them. I still use it every day, and you shouldn't worry about losing any data.

What's the timeline for this? I can't make any promises, but this is what I currently have in mind. I am working on [a recipes manager](https://noeldemartin.com/tasks/implementing-a-recipes-manager-using-solid) that puts some of these new ideas into practice. Once I'm done with that, I will update [Media Kraken](https://github.com/noeldemartin/media-kraken) first, but I don't think that will take me too long. After that, I will update Solid Focus. I only work on this one day a week though, so it'll definitetly take at least some months (I'm not sure if this will be done by 2022).

So yeah, I am aware that this app doesn't work everywhere and there's a lot of room for improvement. And I intend to fix that, no specific dates though! If you want to be notified when it happens, just subscribe to this issue.

**Comment by NoelDeMartin** [comment]
Hey, for anyone following this I've got good news. I finally started working on this! :D

I'm very excited to see how this app will end up now that I know a lot more about Solid, and app development in general. If you're keen to follow the development, I'm working on it in the [aerogel](https://github.com/NoelDeMartin/solid-focus/tree/aerogel) branch and you can read updates here: [Making a Web Application Framework](https://noeldemartin.com/tasks/making-a-web-application-framework).

**Comment by NoelDeMartin** [comment]
Hey, I just published a beta version of the app. You can try it out and give some feedback in this issue: https://github.com/NoelDeMartin/solid-focus/issues/15.

If you want to follow along, or just want to know when the new version is out, you can subscribe to the beta issue. Since I'll be posting the upcoming updates there, I'm closing this issue.
